### PR TITLE
Add wavelength calibration templates to MMTO's Blue Channel and Binospec

### PR DESCRIPTION
**Pull request by tepickering**
This PR started out as adding `reid_arxiv` templates to Binospec for the G600 and G1000 gratings and expanded into doing the same for all of the commonly used gratings for Blue Channel. In the process of testing on various datasets, several bugs were found in the Blue Channel support (e.g. not grouping by grating angle) and some more enhancements were made (e.g. parse arc lamps from the FITS headers). A companion PR adds coverage of these changes to the Dev Suite: https://github.com/pypeit/PypeIt-development-suite/pull/242. 

Running `update_docs` pulled in a lot of unrelated changes. A bug is fixed in `gemini_gmos.rst` that was generating a bunch of warnings. There are four remaining warnings, but I can't figure out why or where they're coming from:
```
include/links.rst:3: WARNING: Bullet list ends without a blank line; unexpected unindent.
include/links.rst:3: WARNING: Bullet list ends without a blank line; unexpected unindent.
include/links.rst:7: WARNING: Bullet list ends without a blank line; unexpected unindent.
include/links.rst:1: WARNING: Bullet list ends without a blank line; unexpected unindent.
```
Happy to fix them here, but also pretty certain they're unrelated to the substantive changes I've made.

### CODE REVIEW

**Comment by kbwestfall** [comment]
@tepickering You can ignore those doc warnings.  I should have fixed them in the NIRES Tutorial PR.  (They were particularly annoying because they had nothing to do with the `links.rst` file!)

File: doc/help/pypeit_cache_github_data.rst
Comment (kbwestfall) [inline doc/help/pypeit_cache_github_data.rst:28]: Is this a python version issue?  I'm running python 3.9.

File: pypeit/spectrographs/mmt_bluechannel.py
Comment (kbwestfall) [inline pypeit/spectrographs/mmt_bluechannel.py]: I would change these `in` conditionals to `==` when there is only one option.  I.e., change to `if ftype == 'science':`.

**Review by kbwestfall — APPROVED** [review]
Looks good to me.  Only a couple of minor things.

File: doc/help/pypeit_cache_github_data.rst
Comment (tepickering) [reply doc/help/pypeit_cache_github_data.rst:28]: oh! good point. i was in my 3.11 env when i ran `update_docs`...

File: pypeit/spectrographs/mmt_bluechannel.py
Comment (tepickering) [reply pypeit/spectrographs/mmt_bluechannel.py]: yeah, i was moving things in and out of lists as i was finding new edge cases in the archived data. i'll clean that up.

Comment (tepickering) [reply pypeit/spectrographs/mmt_bluechannel.py]: that said, the optimal solution is to use `match`-`case` and require python >=3.10...

**Comment by tepickering** [comment]
> @tepickering You can ignore those doc warnings. I should have fixed them in the NIRES Tutorial PR. (They were particularly annoying because they had nothing to do with the `links.rst` file!)

omg, i did some digging to try to figure it out, but soon came to realize that would lead to madness...

File: pypeit/spectrographs/mmt_bluechannel.py
Comment (kbwestfall) [reply pypeit/spectrographs/mmt_bluechannel.py]: Ha!  I had exactly the same thought, but decided against the suggestion :)

Comment (tepickering) [reply pypeit/spectrographs/mmt_bluechannel.py]: it's worth discussing making the move to require 3.10+. 3.11 is consistently at least 5% faster than 3.9 for every reduction i've compared. maybe target the `1.13.0` release for that.

File: pypeit/spectrographs/mmt_bluechannel.py
Comment (jhennawi) [inline pypeit/spectrographs/mmt_bluechannel.py:199]: So currently this says below that the wavelength calibration is done with holy-grail. You overwrite this below in config_specific_par, but I'm wondering what the most transparent approach is here. In other words, do you want to default to holy-grail if config_specific_par does not find a match? I think this does that, and one can debate about how to make that most explicit. But perhaps it would be more transparent to the user if an exception was raised in config_specific_par if the grating setup is not calibrated. Currently it will just run with holy-grail in that case which may or may not work, and if it fails it will throw a cryptic exception.

Comment (jhennawi) [reply pypeit/spectrographs/mmt_bluechannel.py:199]: The desired behavior here sort of depends on how likely such a corner case is to occur.

**Review by jhennawi — APPROVED** [review]
Thanks @tepickering. I have a relatively minor comment on the "holy-grail" method default. You are ready to merge after tests pass. I am not sure whether we need a full dev-suite run here, but the team can help you run that. It is never a bad idea to test the code.

File: pypeit/spectrographs/mmt_bluechannel.py
Comment (tepickering) [reply pypeit/spectrographs/mmt_bluechannel.py:199]: yes, the intention here is to default to `holy-grail` and override in `config_specific_par` if there is an available template. this is how `mmt_binospec` works currently and until this PR gets merged. in reality, the blue channel templates here cover all cases i've found in the archived data going back to 2014. the only missing dispersers are an old 600 l/mm grating and the echellette mode. the former has been replaced by the 832 l/mm grating which is vastly superior in every way. the latter hasn't been used in over 20 years and i don't think was ever re-commissioned after changing the detector format. if for some weird reason someone did choose to use the 600 l/mm grating (and assuming we can find where it's stored), there is reasonable expectation that `holy-grail` would work. it works pretty well out of the box with the 800 and 500 gratings at similar central wavelength settings.

**Pull request merged**
